### PR TITLE
Codebase formatting with ``f-string``

### PR DESCRIPTION
**Pull request by ThibFrgsGmz**
| | |
|:---|:---|
|**_Originating Project/Creator_**| |
|**_Affected Component_**|  |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**|  |
|**_Builds Without Errors (y/n)_**|  |
|**_Unit Tests Pass (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |

---

## Change Description

- Replace calls to ``string.format()`` with ``f-strings``
- Replace the use of the ``%`` string interpolation operator with ``f-strings`` (only ``%s`` operator for the moment)

## Rationale

According to the [Python documentation](https://docs.python.org/3/library/stdtypes.html#printf-style-string-formatting), using the ``%`` operator to format strings can lead to "a variety of quirks that lead to a number of common errors."

[PEP 498](https://peps.python.org/pep-0498/) added F-strings to Python in version 3.6. F-strings are a versatile and powerful method of string formatting. Because the code looks more like the output, they make it shorter and more readable.

## Testing/Review Recommendations

Take a deep breath

## Future Work

Deals with operators other than ``%s`` like ``%d``.

### CODE REVIEW

**Comment by ThibFrgsGmz** [comment]
@LeStarch Should I have split this PR into several to make code review easier?
In retrospect, I realize it's a lot of editing, but since it's mostly string formatting...

**Comment by LeStarch** [comment]
@ThibFrgsGmz Yes, we absolutely need to start making smaller PRs that are easier to review.  I am guilty of this too, as I tend to make fairly involved PRs.

Since this is done, and string edits....I'll just go through it as is.  It will just take more time to review.

**Comment by ThibFrgsGmz** [comment]
@LeStarch OK great, sorry for the burden...
The size of my current pending PRs on ``fprime-tools`` are better, right?

**Comment by LeStarch** [comment]
Yes, much more manageable.

File: src/fprime_gds/executables/run_deployment.py [hunk 1]
```diff
@@ -51,14 +49,12 @@ def find_app(root: Path) -> Path:
             files.append(child)
 
     if len(files) == 0:
-        print("[ERROR] No app found in binary location {}".format(bin_dir))
+        print(f"[ERROR] binary location {bin_dir} does not exist")
```
Comment (LeStarch) [inline src/fprime_gds/executables/run_deployment.py]: This error message changed.

**Review by LeStarch — APPROVED** [review]
Fixed one error message that changed.

File: src/fprime_gds/executables/run_deployment.py (hunk 1 above)
Comment (LeStarch) [reply src/fprime_gds/executables/run_deployment.py]: I went ahead and fixed this one.

**Pull request merged**